zebROS_ws/src/behaviors/src/path_intake_actionlib_server.cpp

```diff
@@ -0,0 +1,165 @@
+#include <ros/ros.h>
+#include <actionlib/server/simple_action_server.h>
+
+#include <behavior_actions/GamePiecePickup.h>
+#include <nav_msgs/Path.h>
+#include <behavior_actions/IntakeAction.h>
+#include <path_follower_msgs/PathAction.h>
+
+// This action lib server will first generate a path with game_piece_path_gen, then run that path using PathAction, then intake a game piece using IntakeAction.
+
+void waitForActionlibServer(T &action_client, double timeout, const std::string &activity);
+
+class PathIntakeAction{
+  private:
+    ros::NodeHandle nh_;
+    actionlib::SimpleActionServer<behaviors_actions::IntakeAction> as_;
+    std::string action_name_;
+    behaviors_actions::PathIntakeFeedback feedback_;
+    behaviors_actions::PathIntakeResult result_;
+
+  public:
+    PathIntakeAction(std::string name):
+      // boost bind is like a callback
+      as(nh), name, boost::bind(&PathIntakeAction::executeCB, this, _1), false),
+      action_name_(name)
+    {
+      as_.start();
+    }
+
+    ~PathIntakeAction(void){
+    }
+
+    void executeCB(const behavior_actions::PathIntakeGoalConstPtr &goal){
+      ros::Rate rate(1);
+      bool success = true;
+      bool preempted = false;
+      bool timed_out = false;
+      feedback_.percentage_done = 0; // not implemented yet
+
+      ROS_INFO_STREAM(action_name_.c.str() << ": Following path and intaking");
+
+      // call path generation server which returns path
+
+      nav_msgs::Path path_;
+
+      ros::ServiceClient game_piece_path_gen_client = nh.serviceClient<behavior_actions::GamePiecePickup>("game_piece_path_gen");
+      behaviors::GamePiecePickup game_piece_path_gen_srv;
+      if(game_piece_path_gen_client.call(game_piece_path_gen_srv)){
+        path_ = game_piece_path_server.response.path;
+      }
```

Comment:
maybe add an `else` that logs an error if the path is not returned using `ROS_ERROR_STREAM`?